rulesets/src/collections.ruleset.yml

```diff
@@ -0,0 +1,136 @@
+rules:
+  ##### General #####
+  sps-no-collection-paging-capability:
+    description: Response bodies from collection endpoints must offer paging capability.
+    severity: error
+    given: $.paths[?(!@property.match(/.*\/\{[^}]+\}$/))].get.responses['200'].content.application/json.schema.properties
+    then:
+      - field: "paging"
+        function: truthy
+      - field: "paging.type"
+        function: pattern
+        functionOptions:
+          match: "object"
+
+  ##### Root Element #####
+  sps-collection-missing-results-array:
+    description: Response bodies must have a root element called results and is an array of objects.
+    severity: error
+    given: $.paths[?(!@property.match(/.*\/\{[^}]+\}$/))].get.responses['200'].content.application/json.schema.properties.results
+    then:
+      - field: type
+        function: pattern
+        functionOptions:
+          match: "array"
+      - field: items.type
+        function: pattern
+        functionOptions:
+          match: "object"
+
+  ##### Pagination #####
+  sps-missing-pagination-query-parameters:
+    description: 'Collection GET endpoints must support pagination using query parameters: limit, offset, cursor.'
+    severity: error
+    given: $.paths[?(!@property.match(/.*\/\{[^}]+\}$/))].get
+    then:
+      - field: parameters
+        function: schema
+        functionOptions:
+          schema:
+            type: array
+            items:
+              type: object
+              properties:
+                name:
+                  type: string
+              required:
+                - name
+            minItems: 3
+            contains:
+              anyOf:
+                - properties:
+                    name:
+                      const: offset
+                - properties:
+                    name:
+                      const: limit
+                - properties:
+                    name:
+                      const: cursor
```

Comment:
do we need to be more specific that you can support offset or cursor, but always limit for example?

Reply:
I have updated the rule so it is expecting a query parameter of `limit` and then a query parameter of `offset` or `cursor`.

The rule will return 2 errors if there is a GET endpoint that is missing the limit parameter and one of the paging methods.

Reply:
NICE!